pkg/sql/internal.go

```diff
@@ -1361,25 +1355,24 @@ func (ief *InternalDB) newInternalExecutorWithTxn(
 		mon:        ief.monitor,
 		memMetrics: ief.memMetrics,
 		extraTxnState: &extraTxnState{
-			txn:                    txn,
-			descCollection:         descCol,
-			jobs:                   new(jobsCollection),
-			schemaChangeJobRecords: make(map[descpb.ID]*jobs.Record),
-			schemaChangerState:     schemaChangerState,
+			txn:                txn,
+			descCollection:     descCol,
+			jobsInfo:           newJobsInfo(),
+			schemaChangerState: schemaChangerState,
 		},
 	}
 	populateMinimalSessionData(sd)
 	ie.sessionDataStack = sessiondata.NewStack(sd)
 
 	commitTxnFunc := func(ctx context.Context) error {
 		defer func() {
-			ie.extraTxnState.jobs.reset()
+			ie.extraTxnState.jobsInfo.reset()
 			ie.releaseSchemaChangeJobRecords()
```

Comment:
This relates to my above comment. IIUC, the `reset` already makes the second function a no-op. This can now just be: `defer ie.extraTxnState.jobs.reset()`